test/Libraries/DynamoPythonTests/PythonEditTests.cs

```diff
@@ -237,9 +305,76 @@ public void InputDynamoDictionary_AsPythonDictionary()
             ViewModel.OpenCommand.Execute(examplePath);
 
             var guid = "490a8d54d0fa4782ae18c81f6eef8306";
+            
+            var nodeModel = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace(guid);
+            var pynode = nodeModel as PythonNode;
+
+            foreach (var pythonEngine in pythonEngineVersionsList)
+            {
+                UpdateEnginePropertyForPythonNode(pynode, pythonEngine);
+
+                ViewModel.HomeSpace.Run();
+
+                AssertPreviewValue(guid,
+                      new List<object> { new Dictionary<string, int> { { "abcd", 123 } }, new List<int> { 1, 2, 3, 4, 5, 6, 7, 8, 9 } });
+            }   
+        }
+
+        [Test]
+        public void PythonGeometryTest()
+        {
+            // open test graph
+            var examplePath = Path.Combine(TestDirectory, @"core\python", "PythonGeometry.dyn");
+            ViewModel.OpenCommand.Execute(examplePath);
+
+            var pythonGUID = "3bcad14ed08642789e08ed2759ef92f3";
+            var lineGUID = "cac9ff74bed14ff285294878fa849cd0";
+            var circleCenterPointGUID = "6bb5d21d7815467db1d2ccfc77713337";
+            var circleRadiusGUID = "88746807ba6d4c81a37dbfb187acca81";
+
+            var nodeModel = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace(pythonGUID);
+            var pynode = nodeModel as PythonNode;
+
+            foreach (var pythonEngine in pythonEngineVersionsList)
+            {
+                UpdateEnginePropertyForPythonNode(pynode, pythonEngine);
+
+                ViewModel.HomeSpace.Run();
 
-            AssertPreviewValue(guid,
-                new List<object> {new Dictionary<string, int> {{"abcd", 123}}, new List<int> {1, 2, 3, 4, 5, 6, 7, 8, 9}});
+                var line = GetPreviewValue(lineGUID) as Line;
+                Assert.AreEqual(line.Length, 5);
+
+                AssertPreviewValue(circleCenterPointGUID, Point.ByCoordinates(0, 0, 0));
+                AssertPreviewValue(circleRadiusGUID, 5);
+            }
+        }
+
+        [Test]
+        public void PythonNodeEnginePropertyChangeTest()
+        {
+            // open test graph
+            var examplePath = Path.Combine(TestDirectory, @"core\python", "pythonEngineTest.dyn");
+            ViewModel.OpenCommand.Execute(examplePath);
+            var pythonGUID = "83a5b1d2-dc58-4d8f-823f-861ac7d565f1";
+
+            var nodeModel = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace(pythonGUID);
+            var pynode = nodeModel as PythonNode;
+
+            foreach (var pythonEngine in pythonEngineVersionsList)
+            {
+                UpdateEnginePropertyForPythonNode(pynode, pythonEngine);
+
+                ViewModel.HomeSpace.Run();
+
+                var nodeValue = GetPreviewValue(pythonGUID);
```

Comment:
What node is this that shows the engine version in its preview?

Reply:
This is just a python node that has OUT=sys.version statement executed in python.